zerver/lib/test_fixtures.py

```diff
@@ -60,13 +88,31 @@ def get_migration_status(**options: Any) -> str:
     output = out.read()
     return re.sub('\x1b\[(1|0)m', '', output)
 
-def are_migrations_the_same(migration_file: str, **options: Any) -> bool:
+def extract_migrations_as_list(migration_status: str) -> List[str]:
+    MIGRATIONS_RE = re.compile('\[[X| ]\] (\d+_.+)\n')
+    return MIGRATIONS_RE.findall(migration_status)
+
+def what_to_do_with_migrations(migration_file: str, **options: Any) -> str:
     if not os.path.exists(migration_file):
-        return False
+        return 'scrap'
 
     with open(migration_file) as f:
-        migration_content = f.read()
-    return migration_content == get_migration_status(**options)
+        previous_migration_status = f.read()
+    current_migration_status = get_migration_status(**options)
+    all_curr_migrations = extract_migrations_as_list(current_migration_status)
+    all_prev_migrations = extract_migrations_as_list(previous_migration_status)
+
+    if len(all_curr_migrations) < len(all_prev_migrations):
+        return 'scrap'
+
+    for migration in all_prev_migrations:
+        if migration not in all_curr_migrations:
+            return 'scrap'
+
+    if len(all_curr_migrations) == len(all_prev_migrations):
+        return 'migrations_are_latest'
```

Comment:
Can this just be a check that they're equal?

Reply:
yeah basically in the previous lines we checked that all the previous migrations are present in the latest migration status
If now there length is same then nothing could have changed.